enhancements/20250203-third-party-storage-support.md

```diff
@@ -0,0 +1,333 @@
+    - GoTo Image Page
+    - Click the VM Image
+    - Click the `Download` button
+    - Check the VM Image is downloaded locally
+  - Delete the VM Image and ensure the golden image is deleted.
+    - GoTo Image Page
+    - Click the VM Image
+    - Click the `Delete` button
+    - Check the golden image is deleted with following command.
+    ```shell
+    // find the golden image name
+    $ imageName=$(kubectl get vmimage -n <namespace> |grep <image display name on UI> |awk '{print $1}')
+    // check the golden image
+    $ kubectl get datavolume -n <namespace> |grep ${imageName}
+    $ kubectl get pvc |grep ${imageName}
+    ```
+    - The above command should not return any result.
+
+- VM Root Disk
+  - Ensure we could boot from the RAW VM Image.
+    - Upload the RAW type VM Image
+    - GoTo Virtual Machines Page
+    - Create VM and choose the above VM Image in the `Volumes` tab
+    - Create and wait for the VM to be ready
+  - Ensure we could boot from the QCOW2 VM Image.
+    - Upload the QCOW2 type VM Image
+    - GoTo Virtual Machines Page
+    - Create VM and choose the above VM Image in the `Volumes` tab
+    - Create and wait for the VM to be ready
+
+- VM Data Disk
+  - Ensure the Data Disk can be created with the VM with the host disk path. (Means configure before VM start)
+    - GoTo Virtual Machines Page
+    - Create VM with any VM Image and Add Volume with above StorageClass for 3rd-party on the `Volumes` tab
+    - Create and wait for the VM to be ready
+  - Ensure the Data Disk can be hotplugged to the VM. (Means configure after VM start)
+    - GoTo Virtual Machines Page
+    - Create VM with any VM Image
+    - Create and wait for the VM to be ready
+    - GoTo Volumes Page
+    - Create Volume with above StorageClass for 3rd-party
+    - GoTo Virtual Machines Page
+    - Click the VM
+    - Click `add volume` button and choose the above Volume to attach
+    - Check the Volume is attached to the VM
+
+- Volume Export To VM Image
+  - Ensure the volume can be exported to VM Image with any 3rd-party storage solution.
+    - GoTo Volumes Page
+    - Choose any Volume
+    - Click the `Export Image` button
+    - Choose any `StorageClass` for the 3rd-party storage solution
+    - Create and wait for the VM Image to be ready
+
+**NOTE**: The volume should be exported to any 3rd-party storage solution but only the longhorn v1 volume can be exported to longhorn v1 VM Image.
+
+- VM Template Generator
+  - Ensure the VM can generate the VM template with various disk configurations. (Like a mix of the Longhorn v1 and LVM)
+    - GoTo Virtual Machines Page
+    - Click the VM
+    - Click the `Generate Template` button and ensure the `With Data` checkbox is checked
+    - Create the VM with the above VM template
+    - Create and wait for the VM to be ready
+
+- VM Live Migration
+  - Ensure the VM can be live migrated with various disk configurations. (Like a mix of the Longhorn v1 and LVM)
```

Comment:
Does LVM support live migration? I thought its specific per node.

Reply:
nice catch. It's a typo.

Updated!